gateway-spi/src/main/java/org/apache/knox/gateway/deploy/ServiceDeploymentContributorBase.java

```diff
@@ -82,7 +83,9 @@ protected void addAuthenticationFilter( DeploymentContext context, Service servi
   protected void addIdentityAssertionFilter( DeploymentContext context, Service service, ResourceDescriptor resource) {
     if( topologyContainsProviderType( context, "authentication" ) ||
         topologyContainsProviderType( context, "federation"  ) ) {
-      context.contributeFilter( service, resource, "identity-assertion", null, null );
+      Topology topology = context.getTopology();
+      Provider activeProvider = topology.getProvider("identity-assertion", null);
```

Comment:
Can `topology` be `null` here? If so, we may want to check it.

Reply:
No, one line above there is a `topologyContainsProviderType` which does a `context.getTopology().getProviders()`. If it was null it would fail even earlier.